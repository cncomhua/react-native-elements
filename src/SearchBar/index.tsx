import { withTheme } from '../config';
import { SearchBar, SearchBarProps } from './SearchBar';
import { SearchBarAndroidProps } from './SearchBar-android';
import { SearchBarIosProps } from './SearchBar-ios';
import { SearchBarDefaultProps } from './SearchBar-default';

export { SearchBar };
export type {
  SearchBarProps,
  SearchBarAndroidProps,
  SearchBarDefaultProps,
  SearchBarIosProps,
};
/**
 * RNE SearchBar
 *
 * **Preview**
 *
 * ![URL](URL)
 *
 * [**Documentation**](https://reactnativeelements.com/docs/searchbar)
 */
export default withTheme(SearchBar, 'SearchBar');
