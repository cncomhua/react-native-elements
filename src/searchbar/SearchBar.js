import React from 'react';
import PropTypes from 'prop-types';

import IOSSearchBar from './SearchBar-ios';
import AndroidSearchBar from './SearchBar-android';
import DefaultSearchBar from './SearchBar-default';
import { merge, ThemeConsumer } from '../config';

const SEARCHBAR_COMPONENTS = {
  ios: IOSSearchBar,
  android: AndroidSearchBar,
  default: DefaultSearchBar,
};

const SearchBar = props => {
  const SearchBar = SEARCHBAR_COMPONENTS[props.platform] || DefaultSearchBar;
  return <SearchBar ref={ref => (props.searchBarRef = ref)} {...props} />;
};

SearchBar.propTypes = {
  searchBarRef: PropTypes.func,
  platform: PropTypes.oneOf(['default', 'ios', 'android']),
};

SearchBar.defaultProps = {
  platform: 'default',
};

export default class ThemedSearchBar extends React.Component {
  focus = () => {
    this.searchbar.focus();
  };

  blur = () => {
    this.searchbar.blur();
  };

  clear = () => {
    this.searchbar.clear();
  };

  cancel = () => {
    this.searchbar.cancel && this.searchbar.cancel();
  };

  render() {
    return (
      <ThemeConsumer>
        {({ theme }) => (
          <SearchBar
            searchBarRef={ref => (this.searchbar = ref)}
            {...merge({}, theme.SearchBar, this.props)}
            theme={theme}
          />
        )}
      </ThemeConsumer>
    );
  }
}
