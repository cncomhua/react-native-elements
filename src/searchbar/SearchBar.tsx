import React from 'react';
import { withTheme } from '../config';
import IOSSearchBar, { SearchBarIosProps } from './SearchBar-ios';
import AndroidSearchBar, { SearchBarAndroidProps } from './SearchBar-android';
import DefaultSearchBar, { SearchBarDefaultProps } from './SearchBar-default';
import {
  ActivityIndicatorProps,
  StyleProp,
  TextStyle,
  ViewStyle,
} from 'react-native';
import { IconNode } from '../icons/Icon';
import { Theme, FullTheme } from '../config/theme';
type RecursivePartial<T> = { [P in keyof T]?: RecursivePartial<T[P]> };
const SEARCHBAR_COMPONENTS = {
  ios: IOSSearchBar,
  android: AndroidSearchBar,
  default: DefaultSearchBar,
};

export type SearchBarBaseProps = {
  platform: 'default' | 'ios' | 'android';
  containerStyle?: StyleProp<ViewStyle>;
  inputContainerStyle?: StyleProp<ViewStyle>;
  clearIcon?: IconNode;
  searchIcon?: IconNode;
  inputStyle?: StyleProp<TextStyle>;
  loadingProps?: ActivityIndicatorProps;
  showLoading?: boolean;
  leftIconContainerStyle?: StyleProp<ViewStyle>;
  rightIconContainerStyle?: StyleProp<ViewStyle>;
  onClear?(): void;
  onFocus?(): void;
  onBlur?(): void;
  onChangeText?(text: string): void;
  onCancel?(): void;
  theme?: Theme;
  updateTheme: (updates: RecursivePartial<FullTheme>) => void;
  replaceTheme: (updates: RecursivePartial<FullTheme>) => void;
};

export type SearchBarProps = SearchBarBaseProps &
  SearchBarDefaultProps &
  SearchBarAndroidProps &
  SearchBarIosProps;

class SearchBar extends React.Component<SearchBarBaseProps> {
  searchbar!: IOSSearchBar;
  static defaultProps = {
    platform: 'default' as const,
  };

  focus = () => {
    this.searchbar.focus();
  };

  blur = () => {
    this.searchbar.blur();
  };

  clear = () => {
    this.searchbar.clear();
  };

  cancel = () => {
    this.searchbar.cancel && this.searchbar.cancel();
  };

  render() {
    const Component: typeof React.Component =
      SEARCHBAR_COMPONENTS[this.props.platform] || DefaultSearchBar;

    return (
      <Component
        // @ts-ignore
        ref={(ref: IOSSearchBar) => {
          this.searchbar = ref;
        }}
        {...this.props}
      />
    );
  }
}

export { SearchBar };
export default withTheme(SearchBar, 'SearchBar');
