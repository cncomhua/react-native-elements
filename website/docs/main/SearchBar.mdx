---
id: searchbar
title: SearchBar
slug: /searchbar
---

import Usage from './usage/SearchBar/SearchBar.mdx';

import Props from './props/SearchBar.mdx';

SearchBars are used to search or filter items. Use a SearchBar when the number
of items directly impacts a user's ability to find one of them.

## Usage

<Usage />

## Props

<Props />
