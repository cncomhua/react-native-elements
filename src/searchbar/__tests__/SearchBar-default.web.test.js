import React from 'react';
import { Text, TouchableHighlight, View } from 'react-native';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import parseSamples, {
  assignRef,
  genRefId,
} from 'enzyme-styleguidist-sample-parser';
import options, {
  snapShot,
  buildJsxForGuideMethod,
  ensureCalled,
  onlyEnsureCalled,
  onlySnapshots,
} from '../../../samples';
import SearchBar from '../SearchBar-default';

const props = {
  'no props': {
    component: SearchBar,
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  containerStyle: {
    component: SearchBar,
    props: {
      containerStyle: { borderWidth: 5, borderColor: '#071' },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  inputStyle: {
    component: SearchBar,
    props: {
      inputStyle: { borderWidth: 5, borderColor: '#071' },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  icon: {
    component: SearchBar,
    props: {
      icon: { name: 'pets', color: '#071' },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  noIcon: {
    component: SearchBar,
    props: {
      noIcon: true,
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  lightTheme: {
    component: SearchBar,
    props: {
      lightTheme: true,
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  round: {
    component: SearchBar,
    props: {
      round: true,
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  underlineColorAndroid: {
    component: SearchBar,
    props: {
      underlineColorAndroid: '#071',
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  loadingIcon: {
    component: SearchBar,
    props: {
      showLoadingIcon: true,
      loadingIcon: { color: '#071' },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  placeholder: {
    component: SearchBar,
    props: {
      placeholder: 'placeholder',
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  placeholderTextColor: {
    component: SearchBar,
    props: {
      placeholder: 'placeholder',
      placeholderTextColor: '#071',
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  onChangeText: {
    component: SearchBar,
    props: {
      onChangeText: text => {
        console.log(text);
      },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  onClearText: {
    component: SearchBar,
    props: {
      defaultValue: 'text to clear',
      clearIcon: { name: 'clear' },
      onClearText: () => {
        console.log('onClearText');
      },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
  clearIcon: {
    component: SearchBar,
    props: {
      showLoadingIcon: true,
      clearIcon: { name: 'clear', color: '#071' },
    },
    enzyme: {
      tests: { shallow: { snapshot: snapShot() } },
    },
    styleguidist: {},
  },
};

const methods = {
  focus: {
    component: SearchBar,
    props: { defaultValue: 'cursor will appear' },
    styleguidist: {
      script: `const View = RN.View;\nconst TouchableHighlight = RN.TouchableHighlight;\nconst Text = RN.Text;\n`,
      cd: 1000,
      buildJsx: buildJsxForGuideMethod,
    },
    enzyme: {
      tests: {
        shallow: {
          'ensure called': ensureCalled(),
        },
      },
    },
  },
  blur: {
    component: SearchBar,
    props: { defaultValue: 'place cursor here and watch it be removed' },
    styleguidist: {
      script: `const View = RN.View;\nconst TouchableHighlight = RN.TouchableHighlight;\nconst Text = RN.Text;\n`,
      cd: 3000,
      buildJsx: buildJsxForGuideMethod,
    },
    enzyme: {
      tests: {
        shallow: {
          'ensure called': ensureCalled(),
        },
      },
    },
  },
  clearText: {
    component: SearchBar,
    props: { defaultValue: 'text to clear' },
    styleguidist: {
      script: `const View = RN.View;\nconst TouchableHighlight = RN.TouchableHighlight;\nconst Text = RN.Text;\n`,
      buildJsx: buildJsxForGuideMethod,
    },
    enzyme: {
      tests: {
        shallow: {
          'ensure called': ensureCalled(),
        },
      },
    },
  },
};
const samples = {
  props,
  methods,
};

parseSamples(
  {
    Input: { sectionComponents: { SearchBar: { samples } } },
  },
  options
);

describe('Search component', () => {
  it('should render without issues', () => {
    const component = shallow(<SearchBar />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with icons', () => {
    const component = shallow(
      <SearchBar
        clearIcon={{
          name: '3d-rotation',
          color: 'red',
        }}
        showLoadingIcon
        loadingIcon={{
          style: { flex: 1 },
        }}
        lightTheme
        containerStyle={{ height: 70 }}
      />
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
  it('should call onTextChange when close icon is touched', () => {
    const onChangeTextMock = jest.fn();
    const component = shallow(
      <SearchBar textInputRef="ti" clearIcon onChangeText={onChangeTextMock} />
    );
    component.find('Icon[name="close"]').simulate('press');
    expect(onChangeTextMock).toBeCalled();
  });

  it('should render without icon', () => {
    const component = shallow(
      <SearchBar underlineColorAndroid="red" noIcon round />
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});
