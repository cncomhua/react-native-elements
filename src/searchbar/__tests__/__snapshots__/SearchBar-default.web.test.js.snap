// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input: SearchBar: Props Input: SearchBar: [ prop: clearIcon ] enzyme-shallow: snapshot: Input: SearchBar: [ prop: clearIcon ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#393e42",
      "borderBottomColor": "#000",
      "borderBottomWidth": 1,
      "borderTopColor": "#000",
      "borderTopWidth": 1,
    }
  }
>
  <TextInput
    autoCapitalize="sentences"
    autoComplete="on"
    autoCorrect={true}
    editable={true}
    keyboardType="default"
    multiline={false}
    numberOfLines={2}
    placeholderTextColor="#86939e"
    secureTextEntry={false}
    showLoadingIcon={true}
    style={
      Object {
        "backgroundColor": "#303337",
        "borderRadius": 3,
        "color": "#86939e",
        "fontSize": 14,
        "height": 40,
        "margin": 8,
        "overflow": "hidden",
        "paddingLeft": 26,
        "paddingRight": 30,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Icon
    color="#86939e"
    name="search"
    size={16}
    style={
      Object {
        "backgroundColor": "transparent",
        "left": 16,
        "position": "absolute",
        "top": 19,
      }
    }
  />
  <Icon
    color="#071"
    name="clear"
    onPress={[Function]}
    size={16}
    style={
      Object {
        "backgroundColor": "transparent",
        "position": "absolute",
        "right": 16,
        "top": 19,
      }
    }
  />
</View>
`;

exports[`Input: SearchBar: Props Input: SearchBar: [ prop: containerStyle ] enzyme-shallow: snapshot: Input: SearchBar: [ prop: containerStyle ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#393e42",
      "borderBottomColor": "#000",
      "borderBottomWidth": 1,
      "borderColor": "#071",
      "borderTopColor": "#000",
      "borderTopWidth": 1,
      "borderWidth": 5,
    }
  }
>
  <TextInput
    autoCapitalize="sentences"
    autoComplete="on"
    autoCorrect={true}
    editable={true}
    keyboardType="default"
    multiline={false}
    numberOfLines={2}
    placeholderTextColor="#86939e"
    secureTextEntry={false}
    style={
      Object {
        "backgroundColor": "#303337",
        "borderRadius": 3,
        "color": "#86939e",
        "fontSize": 14,
        "height": 40,
        "margin": 8,
        "overflow": "hidden",
        "paddingLeft": 26,
        "paddingRight": 19,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Icon
    color="#86939e"
    name="search"
    size={16}
    style={
      Object {
        "backgroundColor": "transparent",
        "left": 16,
        "position": "absolute",
        "top": 19,
      }
    }
  />
</View>
`;

exports[`Input: SearchBar: Props Input: SearchBar: [ prop: icon ] enzyme-shallow: snapshot: Input: SearchBar: [ prop: icon ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#393e42",
      "borderBottomColor": "#000",
      "borderBottomWidth": 1,
      "borderTopColor": "#000",
      "borderTopWidth": 1,
    }
  }
>
  <TextInput
    autoCapitalize="sentences"
    autoComplete="on"
    autoCorrect={true}
    editable={true}
    keyboardType="default"
    multiline={false}
    numberOfLines={2}
    placeholderTextColor="#86939e"
    secureTextEntry={false}
    style={
      Object {
        "backgroundColor": "#303337",
        "borderRadius": 3,
        "color": "#86939e",
        "fontSize": 14,
        "height": 40,
        "margin": 8,
        "overflow": "hidden",
        "paddingLeft": 26,
        "paddingRight": 19,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Icon
    color="#071"
    name="pets"
    size={16}
    style={
      Object {
        "backgroundColor": "transparent",
        "left": 16,
        "position": "absolute",
        "top": 19,
      }
    }
  />
</View>
`;

exports[`Input: SearchBar: Props Input: SearchBar: [ prop: inputStyle ] enzyme-shallow: snapshot: Input: SearchBar: [ prop: inputStyle ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#393e42",
      "borderBottomColor": "#000",
      "borderBottomWidth": 1,
      "borderTopColor": "#000",
      "borderTopWidth": 1,
    }
  }
>
  <TextInput
    autoCapitalize="sentences"
    autoComplete="on"
    autoCorrect={true}
    editable={true}
    keyboardType="default"
    multiline={false}
    numberOfLines={2}
    placeholderTextColor="#86939e"
    secureTextEntry={false}
    style={
      Object {
        "backgroundColor": "#303337",
        "borderColor": "#071",
        "borderRadius": 3,
        "borderWidth": 5,
        "color": "#86939e",
        "fontSize": 14,
        "height": 40,
        "margin": 8,
        "overflow": "hidden",
        "paddingLeft": 26,
        "paddingRight": 19,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Icon
    color="#86939e"
    name="search"
    size={16}
    style={
      Object {
        "backgroundColor": "transparent",
        "left": 16,
        "position": "absolute",
        "top": 19,
      }
    }
  />
</View>
`;

exports[`Input: SearchBar: Props Input: SearchBar: [ prop: lightTheme ] enzyme-shallow: snapshot: Input: SearchBar: [ prop: lightTheme ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#e1e8ee",
      "borderBottomColor": "#e1e1e1",
      "borderBottomWidth": 1,
      "borderTopColor": "#e1e1e1",
      "borderTopWidth": 1,
    }
  }
>
  <TextInput
    autoCapitalize="sentences"
    autoComplete="on"
    autoCorrect={true}
    editable={true}
    keyboardType="default"
    multiline={false}
    numberOfLines={2}
    placeholderTextColor="#86939e"
    secureTextEntry={false}
    style={
      Object {
        "backgroundColor": "#bdc6cf",
        "borderRadius": 3,
        "color": "#86939e",
        "fontSize": 14,
        "height": 40,
        "margin": 8,
        "overflow": "hidden",
        "paddingLeft": 26,
        "paddingRight": 19,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Icon
    color="#86939e"
    name="search"
    size={16}
    style={
      Object {
        "backgroundColor": "transparent",
        "left": 16,
        "position": "absolute",
        "top": 19,
      }
    }
  />
</View>
`;

exports[`Input: SearchBar: Props Input: SearchBar: [ prop: loadingIcon ] enzyme-shallow: snapshot: Input: SearchBar: [ prop: loadingIcon ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#393e42",
      "borderBottomColor": "#000",
      "borderBottomWidth": 1,
      "borderTopColor": "#000",
      "borderTopWidth": 1,
    }
  }
>
  <TextInput
    autoCapitalize="sentences"
    autoComplete="on"
    autoCorrect={true}
    editable={true}
    keyboardType="default"
    loadingIcon={
      Object {
        "color": "#071",
      }
    }
    multiline={false}
    numberOfLines={2}
    placeholderTextColor="#86939e"
    secureTextEntry={false}
    showLoadingIcon={true}
    style={
      Object {
        "backgroundColor": "#303337",
        "borderRadius": 3,
        "color": "#86939e",
        "fontSize": 14,
        "height": 40,
        "margin": 8,
        "overflow": "hidden",
        "paddingLeft": 26,
        "paddingRight": 19,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Icon
    color="#86939e"
    name="search"
    size={16}
    style={
      Object {
        "backgroundColor": "transparent",
        "left": 16,
        "position": "absolute",
        "top": 19,
      }
    }
  />
</View>
`;

exports[`Input: SearchBar: Props Input: SearchBar: [ prop: no props ] enzyme-shallow: snapshot: Input: SearchBar: [ prop: no props ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#393e42",
      "borderBottomColor": "#000",
      "borderBottomWidth": 1,
      "borderTopColor": "#000",
      "borderTopWidth": 1,
    }
  }
>
  <TextInput
    autoCapitalize="sentences"
    autoComplete="on"
    autoCorrect={true}
    editable={true}
    keyboardType="default"
    multiline={false}
    numberOfLines={2}
    placeholderTextColor="#86939e"
    secureTextEntry={false}
    style={
      Object {
        "backgroundColor": "#303337",
        "borderRadius": 3,
        "color": "#86939e",
        "fontSize": 14,
        "height": 40,
        "margin": 8,
        "overflow": "hidden",
        "paddingLeft": 26,
        "paddingRight": 19,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Icon
    color="#86939e"
    name="search"
    size={16}
    style={
      Object {
        "backgroundColor": "transparent",
        "left": 16,
        "position": "absolute",
        "top": 19,
      }
    }
  />
</View>
`;

exports[`Input: SearchBar: Props Input: SearchBar: [ prop: noIcon ] enzyme-shallow: snapshot: Input: SearchBar: [ prop: noIcon ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#393e42",
      "borderBottomColor": "#000",
      "borderBottomWidth": 1,
      "borderTopColor": "#000",
      "borderTopWidth": 1,
    }
  }
>
  <TextInput
    autoCapitalize="sentences"
    autoComplete="on"
    autoCorrect={true}
    editable={true}
    keyboardType="default"
    multiline={false}
    numberOfLines={2}
    placeholderTextColor="#86939e"
    secureTextEntry={false}
    style={
      Object {
        "backgroundColor": "#303337",
        "borderRadius": 3,
        "color": "#86939e",
        "fontSize": 14,
        "height": 40,
        "margin": 8,
        "overflow": "hidden",
        "paddingLeft": 9,
        "paddingRight": 19,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;

exports[`Input: SearchBar: Props Input: SearchBar: [ prop: onChangeText ] enzyme-shallow: snapshot: Input: SearchBar: [ prop: onChangeText ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#393e42",
      "borderBottomColor": "#000",
      "borderBottomWidth": 1,
      "borderTopColor": "#000",
      "borderTopWidth": 1,
    }
  }
>
  <TextInput
    autoCapitalize="sentences"
    autoComplete="on"
    autoCorrect={true}
    editable={true}
    keyboardType="default"
    multiline={false}
    numberOfLines={2}
    onChangeText={[Function]}
    placeholderTextColor="#86939e"
    secureTextEntry={false}
    style={
      Object {
        "backgroundColor": "#303337",
        "borderRadius": 3,
        "color": "#86939e",
        "fontSize": 14,
        "height": 40,
        "margin": 8,
        "overflow": "hidden",
        "paddingLeft": 26,
        "paddingRight": 19,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Icon
    color="#86939e"
    name="search"
    size={16}
    style={
      Object {
        "backgroundColor": "transparent",
        "left": 16,
        "position": "absolute",
        "top": 19,
      }
    }
  />
</View>
`;

exports[`Input: SearchBar: Props Input: SearchBar: [ prop: onClearText ] enzyme-shallow: snapshot: Input: SearchBar: [ prop: onClearText ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#393e42",
      "borderBottomColor": "#000",
      "borderBottomWidth": 1,
      "borderTopColor": "#000",
      "borderTopWidth": 1,
    }
  }
>
  <TextInput
    autoCapitalize="sentences"
    autoComplete="on"
    autoCorrect={true}
    defaultValue="text to clear"
    editable={true}
    keyboardType="default"
    multiline={false}
    numberOfLines={2}
    placeholderTextColor="#86939e"
    secureTextEntry={false}
    style={
      Object {
        "backgroundColor": "#303337",
        "borderRadius": 3,
        "color": "#86939e",
        "fontSize": 14,
        "height": 40,
        "margin": 8,
        "overflow": "hidden",
        "paddingLeft": 26,
        "paddingRight": 30,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Icon
    color="#86939e"
    name="search"
    size={16}
    style={
      Object {
        "backgroundColor": "transparent",
        "left": 16,
        "position": "absolute",
        "top": 19,
      }
    }
  />
  <Icon
    color="#86939e"
    name="clear"
    onPress={[Function]}
    size={16}
    style={
      Object {
        "backgroundColor": "transparent",
        "position": "absolute",
        "right": 16,
        "top": 19,
      }
    }
  />
</View>
`;

exports[`Input: SearchBar: Props Input: SearchBar: [ prop: placeholder ] enzyme-shallow: snapshot: Input: SearchBar: [ prop: placeholder ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#393e42",
      "borderBottomColor": "#000",
      "borderBottomWidth": 1,
      "borderTopColor": "#000",
      "borderTopWidth": 1,
    }
  }
>
  <TextInput
    autoCapitalize="sentences"
    autoComplete="on"
    autoCorrect={true}
    editable={true}
    keyboardType="default"
    multiline={false}
    numberOfLines={2}
    placeholder="placeholder"
    placeholderTextColor="#86939e"
    secureTextEntry={false}
    style={
      Object {
        "backgroundColor": "#303337",
        "borderRadius": 3,
        "color": "#86939e",
        "fontSize": 14,
        "height": 40,
        "margin": 8,
        "overflow": "hidden",
        "paddingLeft": 26,
        "paddingRight": 19,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Icon
    color="#86939e"
    name="search"
    size={16}
    style={
      Object {
        "backgroundColor": "transparent",
        "left": 16,
        "position": "absolute",
        "top": 19,
      }
    }
  />
</View>
`;

exports[`Input: SearchBar: Props Input: SearchBar: [ prop: placeholderTextColor ] enzyme-shallow: snapshot: Input: SearchBar: [ prop: placeholderTextColor ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#393e42",
      "borderBottomColor": "#000",
      "borderBottomWidth": 1,
      "borderTopColor": "#000",
      "borderTopWidth": 1,
    }
  }
>
  <TextInput
    autoCapitalize="sentences"
    autoComplete="on"
    autoCorrect={true}
    editable={true}
    keyboardType="default"
    multiline={false}
    numberOfLines={2}
    placeholder="placeholder"
    placeholderTextColor="#071"
    secureTextEntry={false}
    style={
      Object {
        "backgroundColor": "#303337",
        "borderRadius": 3,
        "color": "#86939e",
        "fontSize": 14,
        "height": 40,
        "margin": 8,
        "overflow": "hidden",
        "paddingLeft": 26,
        "paddingRight": 19,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Icon
    color="#86939e"
    name="search"
    size={16}
    style={
      Object {
        "backgroundColor": "transparent",
        "left": 16,
        "position": "absolute",
        "top": 19,
      }
    }
  />
</View>
`;

exports[`Input: SearchBar: Props Input: SearchBar: [ prop: round ] enzyme-shallow: snapshot: Input: SearchBar: [ prop: round ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#393e42",
      "borderBottomColor": "#000",
      "borderBottomWidth": 1,
      "borderTopColor": "#000",
      "borderTopWidth": 1,
    }
  }
>
  <TextInput
    autoCapitalize="sentences"
    autoComplete="on"
    autoCorrect={true}
    editable={true}
    keyboardType="default"
    multiline={false}
    numberOfLines={2}
    placeholderTextColor="#86939e"
    secureTextEntry={false}
    style={
      Object {
        "backgroundColor": "#303337",
        "borderRadius": 20,
        "color": "#86939e",
        "fontSize": 14,
        "height": 40,
        "margin": 8,
        "overflow": "hidden",
        "paddingLeft": 26,
        "paddingRight": 19,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Icon
    color="#86939e"
    name="search"
    size={16}
    style={
      Object {
        "backgroundColor": "transparent",
        "left": 16,
        "position": "absolute",
        "top": 19,
      }
    }
  />
</View>
`;

exports[`Input: SearchBar: Props Input: SearchBar: [ prop: underlineColorAndroid ] enzyme-shallow: snapshot: Input: SearchBar: [ prop: underlineColorAndroid ] enzyme-shallow: snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#393e42",
      "borderBottomColor": "#000",
      "borderBottomWidth": 1,
      "borderTopColor": "#000",
      "borderTopWidth": 1,
    }
  }
>
  <TextInput
    autoCapitalize="sentences"
    autoComplete="on"
    autoCorrect={true}
    editable={true}
    keyboardType="default"
    multiline={false}
    numberOfLines={2}
    placeholderTextColor="#86939e"
    secureTextEntry={false}
    style={
      Object {
        "backgroundColor": "#303337",
        "borderRadius": 3,
        "color": "#86939e",
        "fontSize": 14,
        "height": 40,
        "margin": 8,
        "overflow": "hidden",
        "paddingLeft": 26,
        "paddingRight": 19,
      }
    }
    underlineColorAndroid="#071"
  />
  <Icon
    color="#86939e"
    name="search"
    size={16}
    style={
      Object {
        "backgroundColor": "transparent",
        "left": 16,
        "position": "absolute",
        "top": 19,
      }
    }
  />
</View>
`;

exports[`Search component should render with icons 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#e1e8ee",
      "borderBottomColor": "#e1e1e1",
      "borderBottomWidth": 1,
      "borderTopColor": "#e1e1e1",
      "borderTopWidth": 1,
      "height": 70,
    }
  }
>
  <TextInput
    autoCapitalize="sentences"
    autoComplete="on"
    autoCorrect={true}
    editable={true}
    keyboardType="default"
    loadingIcon={
      Object {
        "style": Object {
          "flex": 1,
        },
      }
    }
    multiline={false}
    numberOfLines={2}
    placeholderTextColor="#86939e"
    secureTextEntry={false}
    showLoadingIcon={true}
    style={
      Object {
        "backgroundColor": "#bdc6cf",
        "borderRadius": 3,
        "color": "#86939e",
        "fontSize": 14,
        "height": 40,
        "margin": 8,
        "overflow": "hidden",
        "paddingLeft": 26,
        "paddingRight": 30,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Icon
    color="#86939e"
    name="search"
    size={16}
    style={
      Object {
        "backgroundColor": "transparent",
        "left": 16,
        "position": "absolute",
        "top": 19,
      }
    }
  />
  <Icon
    color="red"
    name="3d-rotation"
    onPress={[Function]}
    size={16}
    style={
      Object {
        "backgroundColor": "transparent",
        "position": "absolute",
        "right": 16,
        "top": 19,
      }
    }
  />
</View>
`;

exports[`Search component should render without icon 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#393e42",
      "borderBottomColor": "#000",
      "borderBottomWidth": 1,
      "borderTopColor": "#000",
      "borderTopWidth": 1,
    }
  }
>
  <TextInput
    autoCapitalize="sentences"
    autoComplete="on"
    autoCorrect={true}
    editable={true}
    keyboardType="default"
    multiline={false}
    numberOfLines={2}
    placeholderTextColor="#86939e"
    secureTextEntry={false}
    style={
      Object {
        "backgroundColor": "#303337",
        "borderRadius": 20,
        "color": "#86939e",
        "fontSize": 14,
        "height": 40,
        "margin": 8,
        "overflow": "hidden",
        "paddingLeft": 9,
        "paddingRight": 19,
      }
    }
    underlineColorAndroid="red"
  />
</View>
`;

exports[`Search component should render without issues 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#393e42",
      "borderBottomColor": "#000",
      "borderBottomWidth": 1,
      "borderTopColor": "#000",
      "borderTopWidth": 1,
    }
  }
>
  <TextInput
    autoCapitalize="sentences"
    autoComplete="on"
    autoCorrect={true}
    editable={true}
    keyboardType="default"
    multiline={false}
    numberOfLines={2}
    placeholderTextColor="#86939e"
    secureTextEntry={false}
    style={
      Object {
        "backgroundColor": "#303337",
        "borderRadius": 3,
        "color": "#86939e",
        "fontSize": 14,
        "height": 40,
        "margin": 8,
        "overflow": "hidden",
        "paddingLeft": 26,
        "paddingRight": 19,
      }
    }
    underlineColorAndroid="transparent"
  />
  <Icon
    color="#86939e"
    name="search"
    size={16}
    style={
      Object {
        "backgroundColor": "transparent",
        "left": 16,
        "position": "absolute",
        "top": 19,
      }
    }
  />
</View>
`;
